openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -1987,13 +1989,7 @@ The following additional Credential Issuer metadata parameters are defined for t
 
 * `credential_definition`: REQUIRED. Object containing the detailed description of the Credential type. It consists of at least the following two parameters:
   * `type`: REQUIRED. Array designating the types a certain Credential type supports, according to [@VC_DATA], Section 4.3.
-  * `credentialSubject`: OPTIONAL. Object containing a list of name/value pairs, where each name identifies a claim offered in the Credential. The value can be another such object (nested data structures), or an array of such objects. To express the specifics about the claim, the most deeply nested value MAY be an object that includes the following parameters defined by this specification (other parameters MAY also be used):
-      * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the claim is not included in the issued Credential if the wallet did not request the inclusion of the claim, and/or if the Credential Issuer chose to not include the claim. If the `mandatory` parameter is omitted, the default value is `false`.
-      * `value_type`: OPTIONAL. String value determining the type of value of the claim. Valid values defined by this specification are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image). Other values MAY also be used.
-      * `display`: OPTIONAL. Array of objects, where each object contains display properties of a certain claim in the Credential for a certain language. Below is a non-exhaustive list of valid parameters that MAY be included:
-          * `name`: OPTIONAL. String value of a display name for the claim.
-          * `locale`: OPTIONAL. String value that identifies language of this object represented as language tag values defined in BCP47 [@!RFC5646]. There MUST be only one object for each language identifier.
-* `order`: OPTIONAL. Array of the claim name values that lists them in the order they should be displayed by the Wallet.
+  * `credentialSubject`: OPTIONAL. An array of claims description objects as defined in (#claims-description).
```

Comment:
The use of an array of claim description objects now allows for multiple contradictory descriptions of the same claim, which I believe was not possible before.
E.g.
```
[
   {"path": ["nationalities"], "mandatory": true},
   {"path": ["nationalities"], "mandatory": false}
] 
```
We should clarify how a consumer should react to those cases:
- Reject the array (e.g. reject the authorization request is this happens on inside `authorization_details`)?
- Only consider the first one entry?
- Merge the properties of both entries (which may only make sense for some cases)?
With arrays, these collisions may be a bit more subtle.
E.g. How should this be handled on an authorization request?
```
[
   {"path": ["nationalities", 0], "mandatory": true},
   {"path": ["nationalities", null], "mandatory": false}
] 
```
Does this mean the `nationalities` array can have one or more elements.

Finally, we could have entries suggesting different structure (and not only different properties), such as
```
[
    {"path": ["nationalities", 0], "display": {...}},
    {"path": ["nationalities", "country"], "display": {...}}
]
```
The first entry indicates that `nationalities` is an array. The second entry indicates `nationalities` is an object.

Reply:
The first and second example show points we indeed need to address. The third one should be taken care of by the current processing rules already.

Reply:
I have fixed this in the latest revision.